### ISSUE TITLE
Page all output, and remove --limit options

### ISSUE THREAD

**Issue by JacobEvelyn** [JacobEvelyn/friends#156]
This change would allow the `list` commands to have paged results (like `git log`, or `git log | more`), and eliminate the need for a default `--limit` argument.

Still need to decide whether this should be used for other types of output (graphs, for instance).

**Comment by andypearson** [comment]
This would be amazing! At the moment I hacked it using the following:

```
function fla {
  command friends list activities --limit=10000 $@ | less -R
}
```

**Comment by JacobEvelyn** [comment]
Yeah agreed that it's much better default behavior. At the moment I'm just waiting for my [`tty-pager` PR](https://github.com/piotrmurach/tty-pager/pull/4) to be merged and then I'll add this feature here.

**Comment by JacobEvelyn** [comment]
@andypearson I've got a proof-of-concept branch out (`less-is-more`). It's a little messy (in particular I don't love how some `Introvert` methods return data and others print) but it works. All of the `graph` and `list` commands now have their output paged.

I'd love to hear what you think. In particular:

1. Are there paged commands that should not be paged?
2. Are there non-paged commands that should be paged?
3. Is it annoying to no longer have `--limit` options for `list activities`?
4. Is it annoying to no longer have `--limit` options for `list favorite friends` and `list favorite locations`?
5. Do you notice a performance impact to the paged commands?

**Comment by andypearson** [comment]
Sheesh, 22 days. Sorry it's taken me so long to reply.

Running `bundle exec friends --filename ~/Dropbox/Friends/friends.md list activities` from the `less-is-more` branch I'm seeing a lot of iffy looking colour codes:

```
*snip*
ESC[1m2017-02-22ESC[0m: Dinner at ESC[1;33mRed Squirrel EmporiumESC[0m with ESC[1;35mRachel ClarkeESC[0m
*snip*
```

I use bash in iTerm2, if that makes any difference?

**Comment by JacobEvelyn** [comment]
@andypearson interesting, thanks for the info. Would you mind pasting the output of this command?

```bash
echo "GIT_PAGER: $GIT_PAGER\nPAGER: $PAGER\ncore.pager: `git config --get-all core.pager`\nless: `which less`\nmore: `which more`\ncat: `which cat`\npager: `which pager`"
```

**Comment by andypearson** [comment]
The newlines didn't work, so output looks like:

```
23:21 $ echo "GIT_PAGER: $GIT_PAGER\nPAGER: $PAGER\ncore.pager: `git config --get-all core.pager`\nless: `which less`\nmore: `which more`\ncat: `which cat`\npager: `which pager`"
GIT_PAGER: \nPAGER: \ncore.pager: \nless: /usr/bin/less\nmore: /usr/bin/more\ncat: /bin/cat\npager:
```

**Comment by JacobEvelyn** [comment]
Interesting, thanks. I think I understand the problem. Can you see if this produces correct-looking output?

```bash
PAGER='less -R' bundle exec bin/friends --filename ~/Dropbox/Friends/friends.md list activities
```

**Comment by andypearson** [comment]
@JacobEvelyn yup, that does the trick!

**Comment by JacobEvelyn** [comment]
Just noting that I'm still working on/tinkering with/thinking about this. With recent changes to the `tty-pager` gem I should be able to use that in a vaguely cross-platform way, though I'm not sure if Windows supports `less -RXF` without use of the new Linux subsystem.

**Comment by JacobEvelyn** [comment]
@andypearson it's been a long time coming, but I finally shipped this feature in version 0.40. Let me know how it works for you! Also, note that you can customize which pager is used: https://github.com/JacobEvelyn/friends#a-note-on-output

**Comment by andypearson** [comment]
Oh cool! I'll give it a shot as soon as I can :)

(And, once again, thanks for this awesome bit of kit, my `friends.md` is currently sitting at a cool ~1200 lines).

```
12:56 ~/Dropbox/Friends ⌔ 2.4.2p198 $ fst
Total activities: 684
Total friends: 238
Total locations: 256
Total notes: 0
Total tags: 36
Total time elapsed: 948 days
```
